### ISSUE TITLE
[REVIEW]: Take a moderndive into introductory linear regression with R

### ISSUE THREAD

**Issue by whedon** [openjournals/jose-reviews#115]
**Submitting author:** @rudeboybert (<a href="http://orcid.org/0000-0001-7824-306X">Albert Y. Kim</a>)
**Repository:** <a href="https://github.com/moderndive/moderndive" target ="_blank">https://github.com/moderndive/moderndive</a>
**Version:** v0.5.2
**Editor:** @moorepants
**Reviewer:** @lisamr, @lwjohnst86
**Archive:** <a href="https://doi.org/10.5281/zenodo.5119261" target="_blank">10.5281/zenodo.5119261</a>

## Status

[![status](https://jose.theoj.org/papers/ca30a7168bbc1d583a5feb2cc596758d/status.svg)](https://jose.theoj.org/papers/ca30a7168bbc1d583a5feb2cc596758d)

Status badge code:

```
HTML: <a href="https://jose.theoj.org/papers/ca30a7168bbc1d583a5feb2cc596758d"><img src="https://jose.theoj.org/papers/ca30a7168bbc1d583a5feb2cc596758d/status.svg"></a>
Markdown: [![status](https://jose.theoj.org/papers/ca30a7168bbc1d583a5feb2cc596758d/status.svg)](https://jose.theoj.org/papers/ca30a7168bbc1d583a5feb2cc596758d)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/moderndive/moderndive" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@lisamr & @lwjohnst86, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/jose-reviews/invitations

The reviewer guidelines are available here: https://openjournals.readthedocs.io/en/jose/reviewer_guidelines.html. Any questions/concerns please let @moorepants know.

✨ **Please try and complete your review in the next four weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/jose/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/moderndive/moderndive">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** v0.5.2
- [x] **Authorship:** Has the submitting author (@rudeboybert) made substantial contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation? (and documentation is sufficient?)
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state the need for this software and who the target audience is?
- [x] **Installation instructions:** Is there a clearly stated list of dependencies? (Ideally these should be handled with an automated package management solution.)
- [x] **Example usage:** Do the authors include examples of how to use the software?
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state the need for this software and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/jose/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/moderndive/moderndive">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** v0.5.2
- [x] **Authorship:** Has the submitting author (@rudeboybert) made substantial contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation? (and documentation is sufficient?)
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state the need for this software and who the target audience is?
- [x] **Installation instructions:** Is there a clearly stated list of dependencies? (Ideally these should be handled with an automated package management solution.)
- [x] **Example usage:** Do the authors include examples of how to use the software?
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state the need for this software and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @lisamr, @lwjohnst86 it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/jose-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/jose-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.83 s (119.8 files/s, 22799.4 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
HTML                            36           1775            445           7021
SVG                             20              0              1           3457
Markdown                         7            439              0           1306
R                               21            291           1059           1129
CSS                              3             99             48            428
JavaScript                       4             65             35            267
Rmd                              2            214            322            176
YAML                             4             19              1            155
TeX                              1              9              1             71
Bourne Shell                     1              1              4              1
-------------------------------------------------------------------------------
SUM:                            99           2912           1916          14011
-------------------------------------------------------------------------------


Statistical information for the repository '3d5a73c19e8a4353347590c3' was
gathered on 2021/05/06.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
ismayc                           3           187              2           29.58
rudeboybert                      7           316            134           70.42

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Chester Ismay                70          100.0         33.1                7.14
rudeboybert                 297           94.0         15.6               10.10
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00115/jose.00115/10.21105.jose.00115.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00115/jose.00115/10.21105.jose.00115.pdf) :page_facing_up: :point_left:

**Comment by moorepants** [comment]
@whedon set v0.5.1 as version

**Comment by whedon** [comment]
OK. v0.5.1 is the version.

**Comment by moorepants** [comment]
@lisamr & @lwjohnst86, the review can now officially start here. Please read from the top of the thread and be sure to review the reviewer guidelines. You'll work your way through the checklists above over the next 4 weeks. Post comments or issues to the main repository as you go along or all at once, your preference. An automatic notice will be sent out in 2 weeks as a reminder. You should review the paper, software documentation, and test out & use the software. The paper authors can (and are encouraged to) interact while you do the review. Let me know if there are any questions. Welcome to JOSE!

**Comment by moorepants** [comment]
Also note that this is should be reviewed as a "educational software tools" submission.

**Comment by lisamr** [comment]
@moorepants I tried to follow this link: "Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations", but I get this message: "Sorry, we couldn't find that repository invitation. It is possible that the invitation was revoked or that you are not logged into the invited account." I'm assuming that's where I can edit the checklist, correct?

**Comment by lisamr** [comment]
I started going through the checklist pasted at the top of this thread:

# General checks
- authorship: It is currently unclear how Max Kuhn contributed, as they are not listed as a contributor on the Github repository and there isn't an author contribution section in the manuscript to clarify. 

# Documentation
- Statement of need: It's missing in the Readme file and in the MS
- Installation instructions: I couldn't find a clearly stated list of dependencies in the repository, but maybe I was looking in the wrong place. I checked under the "dependency graph" tab by clicking on the "insight" button, and no dependencies were listed there either. 

## Minor details on the readme file
 - Under Demo, it's stated 

> Among the many useful features of the moderndive package outlined in our essay “Why should you use the moderndive package for intro linear regression?” we highlight three functions in particular as covered there. We also mention the geom_parallel_slopes() function as #4.  

Just say there are 4 functions covered. I'm not sure why `geom_parallel_slopes` is singled out.
- for the function `get_regression_table()` I suggest at least in the help documentation, it should be stated what the alpha level is. I know it's probably 0.05, but it should be stated, or even allowed to vary since the point of the function is to understand confidence intervals between a given range.
- for the function `get_regression_points()`: If this package is aimed at students very new to statistics, it might be helpful to include the following in the readme.md file: "...including fitted/predicted values (e.g. 'score_hat')"
- for the function `get_regression_summaries()`: excuse me if I'm wrong since for the life of me, I never remembered how to calculate degrees of freedom, but should `df = 2`? The output from `summary(score_model)` shows 1 and 461 DF.

**Comment by moorepants** [comment]
> @moorepants I tried to follow this link: "Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations", but I get this message: "Sorry, we couldn't find that repository invitation. It is possible that the invitation was revoked or that you are not logged into the invited account." I'm assuming that's where I can edit the checklist, correct?

@labarba, do the reviewers have to follow this link? It seems like a joss thing that hasnt' converted to jose.

**Comment by moorepants** [comment]
> Functional documentation and tests: I'm not qualified to assess this since I don't write packages (I'm more of a user than developer)

@lisamr, is it possible for your to try running the tests? It's good if each reviewer can do so. You don't have to inspect the code for them.

**Comment by moorepants** [comment]
Also, each reviewer should be able to have a look at this too:

> Functionality documentation: Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?

This is checking to see if there are understandable vignettes for each public function, class, etc.

**Comment by moorepants** [comment]
Actually, vignettes may have not been precise enough. There is object documentation, see:

https://r-pkgs.org/man.html#man

and vignettes:

https://r-pkgs.org/vignettes.html#vignettes

Ideally, the package will have both of these types of documentation. The first is the "API method documentation" mentioned in the checklist. The second is a bit more long form explanations that aren't tied to a specific object. The object documentation should be present for each public function and the vignettes should be present as per the reviewers' suggestions, mostly to conform with best practices in R documentation.

**Comment by rudeboybert** [comment]
Hi @lisamr thank you for bringing up those checklist items. I'll make the necessary changes and have Whedon generate a new pdf today.

**Comment by lisamr** [comment]
@moorepants got it, so the "functionality documentation" checklist item is referring to the object documentation. @rudeboybert below is my feedback:
- documentation for various datasets look great--good detail on explaining each column and providing source of the data
- `geom_categorical_model()`: It is not obvious how the solid and dashed lines differ or how they were determined and it's not clearly stated in the documentation. I thought maybe pairwise contrasts were being performed, but when I look at two categories that are not significantly different (e.g. `ggplot(evals, aes(pic_outfit, score )) + geom_point() + geom_categorical_model()`), their lines are still different. Then I thought maybe each category gets its own linetype, but when I look at `ggplot(mpg, aes(x = drv, y = hwy)) + geom_point() + geom_categorical_model()`, again there are only 2 types. 
- `get_regression_summaries()`: All of the columns are self-explanatory except for df. More details under the "Value" section would be nice.
- `get_regression_table()`: Again, either state that the upper and lower bounds represent the 95% confidence interval, or even better, allow the interval ranges to vary, which in my opinion will help break down this (artificial) idea that 95% CI and the associated p-values are all that matter. 

# Larger structural comments on the paper
I went through the paper last night and my first impression was that it was long (~2500 words of text) and overlapped quite a lot with the text and code in the readme file. From what I understand, "Detailed documentation should be present in the repository of the submitted software or module, is reviewed there, and does not need to be included in the paper" (link [here](https://openjournals.readthedocs.io/en/jose/review_criteria.html)). It also lacked a Statement of Need section. I thought a lot of the information presented could have been summed up in a shorter summary paragraph. I also thought the code/output could have been condensed by directing the reader to the repository's documentation.

Here are some key descriptions of the paper, quoted from https://openjournals.readthedocs.io/en/jose/submitting.html:
- Write a short Markdown file titled paper.md with title, author names and affiliations, containing a description of the software, statement of need, and key references.
- JOSE papers contain a limited set of metadata, plus Summary & Reference sections. We explicitly do not publish long-form articles, because the scholarship represented by a JOSE article is contained in the software or learning modules themselves. Expected length is around **1000 words max**.

@moorepants lmk if I'm misunderstanding anything or maybe clarify how the software paper should differ from the readme and other vignettes?

**Comment by moorepants** [comment]
> @moorepants lmk if I'm misunderstanding anything or maybe clarify how the software paper should differ from the readme and other vignettes?

Yes, you've interpreted correctly. The documentation and paper should be different as you've described.

**Comment by rudeboybert** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00115/jose.00115/10.21105.jose.00115.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00115/jose.00115/10.21105.jose.00115.pdf) :page_facing_up: :point_left:

**Comment by rudeboybert** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00115/jose.00115/10.21105.jose.00115.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00115/jose.00115/10.21105.jose.00115.pdf) :page_facing_up: :point_left:

**Comment by rudeboybert** [comment]
Hi, I had almost finished addressing the initial comments by @lisamr when they followed up with more feedback. I will address the initial comments only here:

> I started going through the checklist pasted at the top of this thread:
> 
> # General checks
> * authorship: It is currently unclear how Max Kuhn contributed, as they are not listed as a contributor on the Github repository and there isn't an author contribution section in the manuscript to clarify.

I added a "Author contributions" section towards the end of the paper.
> # Documentation
> * Statement of need: It's missing in the Readme file and in the MS

A "Statement of need" section has been added at the outset of the manuscript. 
> * Installation instructions: I couldn't find a clearly stated list of dependencies in the repository, but maybe I was looking in the wrong place. I checked under the "dependency graph" tab by clicking on the "insight" button, and no dependencies were listed there either. https://github.com/moderndive/moderndive/network/dependencies

All the dependencies are automatically installed when installing the `moderndive` package, the list of all dependencies can be seen in the [`DESCRIPTION`](https://github.com/moderndive/moderndive/blob/master/DESCRIPTION) file
> * Functional documentation and tests: I'm not qualified to assess this since I don't write packages (I'm more of a user than developer). @lwjohnst86 do you want to make sure to cover that component?

@moorepants addressed this above
> 
> ## Minor details on the readme file
> * Under Demo, it's stated
> 
> > Among the many useful features of the moderndive package outlined in our essay “Why should you use the moderndive package for intro linear regression?” we highlight three functions in particular as covered there. We also mention the geom_parallel_slopes() function as #4.
> 
> Just say there are 4 functions covered. I'm not sure why `geom_parallel_slopes` is singled out.

That was a typo: The README has been updated to indicate that there are 4 functions highlighted.
> 
> * for the function `get_regression_table()` I suggest at least in the help documentation, it should be stated what the alpha level is. I know it's probably 0.05, but it should be stated, or even allowed to vary since the point of the function is to understand confidence intervals between a given range.

Thank you for this great suggestion. We updated the functionality of this function to allow for a `conf.level` argument in this [commit](https://github.com/moderndive/moderndive/pull/104/commits/12a972f48f9d092475be5cbebf2ac9956c807e96), and updated the README and manuscript accordingly. This is only available in the development version of the package `v0.5.1.9000` for now, I'll wait for more feedback before updating on CRAN. @moorepants could you update the version to `v0.5.1.9000`?
> * for the function `get_regression_points()`: If this package is aimed at students very new to statistics, it might be helpful to include the following in the readme.md file: "...including fitted/predicted values (e.g. 'score_hat')"

Thank you, similar language was added to the README in this [commit](https://github.com/moderndive/moderndive/pull/104/commits/f0a775d7758f17ba0c538c44ab309485c1f79a6f)
> * for the function `get_regression_summaries()`: excuse me if I'm wrong, but should `df = 2`? The output from `summary(score_model)` shows 1 and 461 DF.

You are correct. This seems to have been a bug in the `broom::glance()` function that has since been fixed; the README has been updated.

**Comment by rudeboybert** [comment]
Here are my initial responses to the second round of feedback from @lisamr 

> @moorepants got it, so the "functionality documentation" checklist item is referring to the object documentation. @rudeboybert below is my feedback:
> 
> * documentation for various datasets look great--good detail on explaining each column and providing source of the data

Thank you
> * `geom_categorical_model()`: It is not obvious how the solid and dashed lines differ or how they were determined and it's not clearly stated in the documentation. I thought maybe pairwise contrasts were being performed, but when I look at two categories that are not significantly different (e.g. `ggplot(evals, aes(pic_outfit, score )) + geom_point() + geom_categorical_model()`), their lines are still different. Then I thought maybe each category gets its own linetype, but when I look at `ggplot(mpg, aes(x = drv, y = hwy)) + geom_point() + geom_categorical_model()`, again there are only 2 types.
> * `get_regression_summaries()`: All of the columns are self-explanatory except for df. More details under the "Value" section would be nice.

I will address the above two points in my next update to the manuscript.
> * `get_regression_table()`: Again, either state that the upper and lower bounds represent the 95% confidence interval, or even better, allow the interval ranges to vary, which in my opinion will help break down this (artificial) idea that 95% CI and the associated p-values are all that matter.

Thanks for this great point. See my comment on the updated `get_regression_points()` function in my previous messages.
> 
> # Larger structural comments on the paper
> I went through the paper last night and my first impression was that it was long (~2500 words of text) and overlapped quite a lot with the text and code in the readme file. From what I understand, "Detailed documentation should be present in the repository of the submitted software or module, is reviewed there, and does not need to be included in the paper" (link [here](https://openjournals.readthedocs.io/en/jose/review_criteria.html)). It also lacked a Statement of Need section. I thought a lot of the information presented could have been summed up in a shorter summary paragraph. I also thought the code/output could have been condensed by directing the reader to the repository's documentation.

Thank for this feedback. I have addressed the Statement of Need section in my previous message. I will address your feedback on article length in my next update. 
> 
> Here are some key descriptions of the paper, quoted from https://openjournals.readthedocs.io/en/jose/submitting.html:
> 
> * Write a short Markdown file titled paper.md with title, author names and affiliations, containing a description of the software, statement of need, and key references.
> * JOSE papers contain a limited set of metadata, plus Summary & Reference sections. We explicitly do not publish long-form articles, because the scholarship represented by a JOSE article is contained in the software or learning modules themselves. Expected length is around **1000 words max**.

I somehow missed this guideline (perhaps it was updated after my initial submission last year?) In any case, thanks for putting up with its current length; I will address this in my next update, pending comments from @lwjohnst86.
> 
> @moorepants lmk if I'm misunderstanding anything or maybe clarify how the software paper should differ from the readme and other vignettes?

**Comment by whedon** [comment]
:wave: @lwjohnst86, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @lisamr, please update us on how your review is going (this is an automated reminder).

**Comment by lisamr** [comment]
> 👋 @lisamr, please update us on how your review is going (this is an automated reminder).

I've gone through the checklist as much as I can and I'm actually still waiting on the invite to this URL: https://github.com/openjournals/joss-reviews/invitations. The next major steps are to review the newly shortened version of the paper.

**Comment by moorepants** [comment]
> I'm actually still waiting on the invite to this URL: https://github.com/openjournals/joss-reviews/invitations.

@lisamr I don't think this is important. Don't worry about it. The link is broken as it is for joss, not jose.

**Comment by moorepants** [comment]
> I've gone through the checklist as much as I can 

You can check the boxes at the top for the things you've approved as complete.

**Comment by lisamr** [comment]
> > I've gone through the checklist as much as I can
> 
> You can check the boxes at the top for the things you've approved as complete.

Hmm, I'm afraid I don't know how to edit the checklist up top. Help?

**Comment by labarba** [comment]
@whedon re-invite @lisamr as reviewer

**Comment by whedon** [comment]
The reviewer already has a pending invite.

@lisamr please accept the invite by clicking this link: https://github.com/openjournals/jose-reviews/invitations

**Comment by moorepants** [comment]
There's the correct link! Thanks @labarba.

**Comment by moorepants** [comment]
> Hmm, I'm afraid I don't know how to edit the checklist up top. Help?

Presumably when you accept the new link invite you'll be able to check the boxes.

**Comment by lisamr** [comment]
same issue with the new link: "Sorry, we couldn't find that repository invitation. It is possible that the invitation was revoked or that you are not logged into the invited account."

**Comment by labarba** [comment]
@openjournals/dev Help! 
We have a reviewer unable to edit the checkboxes. Can you check the issue?

**Comment by arfon** [comment]
@whedon re-invite @lisamr as reviewer

**Comment by whedon** [comment]
I'm sorry @arfon, I'm afraid I can't do that. That's something only editors are allowed to do.

**Comment by arfon** [comment]
OK, could you try this link again @lisamr: https://github.com/openjournals/jose-reviews/invitations

**Comment by lisamr** [comment]
@arfon perfect, that link worked and I can finally check the checkboxes above. Thank you!

**Comment by lwjohnst86** [comment]
@rudeboybert @labarba @lisamr I finished my review. All my issues and feedback were either created as Issues or PRs in the moderndive package repo. Overall looks great! If the issues and PRs are addressed, I would say it's good to go!

**Comment by rudeboybert** [comment]
Wow @lwjohnst86 thanks for the very thorough and thoughtful reviews, and double thanks for communicating them as issues & PRs directly in the moderndive package repo! 

I hope to address @lisamr's outstanding comments and Luke's PR's issues next week. I'll tag you both as well as @moorepants when I'm done. Many thanks!

**Comment by moorepants** [comment]
@lisamr and @lwjohnst86 Thanks for your timely and thorough reviews!

@rudeboybert sounds like you are on top of things, looking forward to your responses and adjustments from the feedback.

As was noted by both reviewers, please adjust the paper to fit the JOSE guidelines more closely. It should not be a repeat of the documentation. All other comments look good to me.

**Comment by moorepants** [comment]
@lisamr and @lwjohnst86 Thanks for your timely and thorough reviews!

@rudeboybert sounds like you are on top of things, looking forward to your responses and adjustments from the feedback.

As was noted by both reviewers, please adjust the paper to fit the JOSE guidelines more closely. It should not be a repeat of the documentation. All other comments look good to me.

**Comment by rudeboybert** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00115/jose.00115/10.21105.jose.00115.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00115/jose.00115/10.21105.jose.00115.pdf) :page_facing_up: :point_left:

**Comment by rudeboybert** [comment]
Hello @moorepants @lisamr @lwjohnst86, many thanks for all your work in this review. I believe I've addressed all the outstanding issues, in particular cutting down the manuscript length from ~2500 to 1026 words. Please let me know if I need to do anything else.

**Comment by rudeboybert** [comment]
Hi @moorepants, sounds good. I've broken down my response in 4 sections:

1. Responses (indicating changes) to first round of comments by @lisamr on 5/13
1. Responses (indicating changes) to second round of comments by @lisamr on 5/14
1. Responses (indicating changes) to comments by @lwjohnst86 on 6/4
1. Explanations for unchecked boxes in checklist

# 1. Responses to first round of comments by @lisamr on 5/13

> # General checks
> * authorship: It is currently unclear how Max Kuhn contributed, as they are not listed as a contributor on the Github repository and there isn't an author contribution section in the manuscript to clarify.

I added a "Author contributions" section towards the end of the paper.
> # Documentation
> * Statement of need: It's missing in the Readme file and in the MS

A "Statement of need" section has been added at the outset of the manuscript. 
> * Installation instructions: I couldn't find a clearly stated list of dependencies in the repository, but maybe I was looking in the wrong place. I checked under the "dependency graph" tab by clicking on the "insight" button, and no dependencies were listed there either. https://github.com/moderndive/moderndive/network/dependencies

All the dependencies are automatically installed when installing the `moderndive` package, the list of all dependencies can be seen in the [`DESCRIPTION`](https://github.com/moderndive/moderndive/blob/master/DESCRIPTION) file
> * Functional documentation and tests: I'm not qualified to assess this since I don't write packages (I'm more of a user than developer). @lwjohnst86 do you want to make sure to cover that component?

@lwjohnst86 checked this checkbox.
> 
> ## Minor details on the readme file
> * Under Demo, it's stated
> 
> > Among the many useful features of the moderndive package outlined in our essay “Why should you use the moderndive package for intro linear regression?” we highlight three functions in particular as covered there. We also mention the geom_parallel_slopes() function as #4.
> 
> Just say there are 4 functions covered. I'm not sure why `geom_parallel_slopes` is singled out.

That was a typo: The README has been updated (in the Basic Usage section) to highlight 4 functions, not 3.
> 
> * for the function `get_regression_table()` I suggest at least in the help documentation, it should be stated what the alpha level is. I know it's probably 0.05, but it should be stated, or even allowed to vary since the point of the function is to understand confidence intervals between a given range.

Thank you for this great suggestion. We updated the functionality of this function to allow for a `conf.level` argument in this [commit](https://github.com/moderndive/moderndive/pull/104/commits/12a972f48f9d092475be5cbebf2ac9956c807e96), and updated the README and manuscript accordingly. This is only currently available in the development version of the package `v0.5.1.9000` for now. This will be included in the next `v0.5.2.` version on CRAN.

> * for the function `get_regression_points()`: If this package is aimed at students very new to statistics, it might be helpful to include the following in the readme.md file: "...including fitted/predicted values (e.g. 'score_hat')"

Thank you, similar language was added to the README in this [commit](https://github.com/moderndive/moderndive/pull/104/commits/f0a775d7758f17ba0c538c44ab309485c1f79a6f)
> * for the function `get_regression_summaries()`: excuse me if I'm wrong, but should `df = 2`? The output from `summary(score_model)` shows 1 and 461 DF.

You are correct. This seems to have been a bug in the `broom::glance()` function that has since been fixed; the README has been updated. 


# 2. Responses to second round of comments by @lisamr on 5/14

> @moorepants got it, so the "functionality documentation" checklist item is referring to the object documentation. @rudeboybert below is my feedback:
> 
> * documentation for various datasets look great--good detail on explaining each column and providing source of the data

Thank you
> * `geom_categorical_model()`: It is not obvious how the solid and dashed lines differ or how they were determined and it's not clearly stated in the documentation. I thought maybe pairwise contrasts were being performed, but when I look at two categories that are not significantly different (e.g. `ggplot(evals, aes(pic_outfit, score )) + geom_point() + geom_categorical_model()`), their lines are still different. Then I thought maybe each category gets its own linetype, but when I look at `ggplot(mpg, aes(x = drv, y = hwy)) + geom_point() + geom_categorical_model()`, again there are only 2 types.

Since this function is not used in the README nor is it used in the manuscript, I think it's beyond the scope of this review and thus this point can be left unaddressed. Please let me know if you disagree.

> * `get_regression_summaries()`: All of the columns are self-explanatory except for df. More details under the "Value" section would be nice.

The following note is now included in the README in the "Produce metrics on the quality of regression model fits":  and the degrees of freedom `df`

> * `get_regression_table()`: Again, either state that the upper and lower bounds represent the 95% confidence interval, or even better, allow the interval ranges to vary, which in my opinion will help break down this (artificial) idea that 95% CI and the associated p-values are all that matter.

Thanks for this great point. This was addressed above.

> 
> # Larger structural comments on the paper
> I went through the paper last night and my first impression was that it was long (~2500 words of text) and overlapped quite a lot with the text and code in the readme file. From what I understand, "Detailed documentation should be present in the repository of the submitted software or module, is reviewed there, and does not need to be included in the paper" (link [here](https://openjournals.readthedocs.io/en/jose/review_criteria.html)). It also lacked a Statement of Need section. I thought a lot of the information presented could have been summed up in a shorter summary paragraph. I also thought the code/output could have been condensed by directing the reader to the repository's documentation.
> 
> Here are some key descriptions of the paper, quoted from https://openjournals.readthedocs.io/en/jose/submitting.html:
> 
> * Write a short Markdown file titled paper.md with title, author names and affiliations, containing a description of the software, statement of need, and key references.
> * JOSE papers contain a limited set of metadata, plus Summary & Reference sections. We explicitly do not publish long-form articles, because the scholarship represented by a JOSE article is contained in the software or learning modules themselves. Expected length is around **1000 words max**.
> 
> @moorepants lmk if I'm misunderstanding anything or maybe clarify how the software paper should differ from the readme and other vignettes?

Thanks for this feedback. I have addressed the Statement of Need section in my previous message. Furthermore, the main paper has been cut from ~2500 words to 1026 words with a lot of the content being moved to the repo README


# 3. Responses to comments by @lwjohnst86 on 6/4

> @rudeboybert @labarba @lisamr I finished my review. All my issues and feedback were either created as Issues or PRs in the moderndive package repo. Overall looks great! If the issues and PRs are addressed, I would say it's good to go!

Thanks for filing all your comments and suggestions as GitHub issues and PR's. I will summarize what I did to address all the above referenced issues from 6/4:

- Infrastructure to build website https://github.com/moderndive/moderndive/issues/105: **Implemented and merged**
- Vignette images not showing https://github.com/moderndive/moderndive/issues/106: **Fixed and merged**
- Code coverage was not actually being generated, fix with this action https://github.com/moderndive/moderndive/issues/107: **Fixed and merged**
- Better navigating reference index objects by grouping them (for website) https://github.com/moderndive/moderndive/issues/108: **Fixed and merged**
- Removing 'judgemental' language from text https://github.com/moderndive/moderndive/issues/109: **Removed such language and merged**
- Remove examples from dataset documentations https://github.com/moderndive/moderndive/issues/110: **Removed all example code from dataset documentation. Now package build and check are much quicker. Issue closed**
- Remove use of patchwork from vignette https://github.com/moderndive/moderndive/issues/111: **Thanks for suggestion, but we really want the two plots side-by-side in order to facilitate visual model selection. Closed**
- Strongly encourage using "fitted" or "predicted" compared to "*_hat" https://github.com/moderndive/moderndive/issues/112: **While this is a good idea, it would cause the package to be misaligned with the book, both the print and online edition at moderndive.com. Closed**
- (infrastructure, optional comment) Consider moving dataset documentation into one file https://github.com/moderndive/moderndive/issues/113: **Great idea, done. Close**
- (Infrastructural, optional) Consider using markdown to write Roxygen docs https://github.com/moderndive/moderndive/issues/114: **Another great suggestion which we implemented. Closed**
- Larger comments on paper.md https://github.com/moderndive/moderndive/issues/115. **The main paper has been cut from ~2500 words to 1026 words with a lot of the content being moved to the repo README**
- General tidy up and removing dependencies https://github.com/moderndive/moderndive/issues/116: **Implemented and merged.**
- Repo needs a CONTRIBUTING guidelines file https://github.com/moderndive/moderndive/issues/117: **Added to repo. Closed**
- Add one or two sentences on purpose and reason for package being created in README https://github.com/moderndive/moderndive/issues/118: **Added in Overview section of README. Closed**


# 4. Explanations for unchecked boxes in checklist

## Unchecked boxes for @lisamr 

> A statement of need: Do the authors clearly state the need for this software and who the target audience is?

Added to manuscript

> Functionality documentation: Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?

I've addressed all suggestions to clarify points of confusion in help files (where appropriate) or in the README (for more verbose explanations).

## Unchecked boxes for @lwjohnst86 

> A statement of need: Do the authors clearly state the need for this software and who the target audience is?

Added to manuscript
> Community guidelines: Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

[Added](https://github.com/moderndive/moderndive/blob/master/CONTRIBUTING.md) to main repo. 
> References: Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

All books and articles have DOI links. DOI links for

**Comment by moorepants** [comment]
Thank you for the thorough and clear write up and addressing each point brought up by the reviewers.

@lisamr and @lwjohnst86, can you both take a second pass here at the changes made and report on whether this addresses your prior comments? Please check off the relevant boxes in the checklist too, if addressed.

Please indicate how much time you'd like to finalize your review.

**Comment by lisamr** [comment]
@rudeboybert great job at tending to the comments and cutting down the MS. I have two small remaining comments that if resolved, I think would mean the paper is good to go.

- I still haven't checked the box for Documentation:Statement of Need. My understanding is that you need a Statement of Need in both the README and the software paper. Otherwise, there wouldn't be a checkbox for both (under Documentation and Software paper). 
- In regard to my comment on `geom_categorical_model`: It's possible that the checkbox for "Functionality documentation" solely encompasses content in the README and software paper. If that's the case, you're fine, and I've actually just checked the box for that already. That said, for a potential user of this package, the function documentation for `geom_categorical_model` still leaves room for confusion. For the general utility of this package, a short sentence about why some lines are dashed vs. solid would be nice.

**Comment by rudeboybert** [comment]
Hi @lisamr,

1. I just added the Statement of Need to the README: https://github.com/moderndive/moderndive#statement-of-need
2. I added a note to the documentation for `geom_categorical_model()` explaining what the solid and dashed lines are in this [commit](https://github.com/moderndive/moderndive/commit/d52a4aef3b234b950095703af3a02ee3b4102b53)

Many thanks for all your work on this review

**Comment by lisamr** [comment]
@rudeboybert looks great to me!

**Comment by lwjohnst86** [comment]
@rudeboybert I've checked all the boxes though you still have almost identical text ("six features") between your vignette and your README which I think you should de-duplicate, or rather "don't repeat yourself" :wink: Other than that, it looks great!! :grin:

**Comment by rudeboybert** [comment]
Thanks @lwjohnst86, I will keep the six features text to form a bridge between the README and the manuscript, even at the cost of repeating myself.

@lisamr could you check off the remaining checkbox? After that point I believe @moorepants can move the review to the next phase

**Comment by lisamr** [comment]
@rudeboybert sorry about that, thought I already checked it off.

**Comment by moorepants** [comment]
@rudeboybert I've checked over everything, including @lisamr and @lwjohnst86's final sign off and it all looks ready to go!

Next steps:

1. Create a new release for your github repository and report the tag string here.
2. Create a Zenodo archive of the release made in point 1 and report the DOI here.

**Comment by moorepants** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.5070/T573020084 is OK
- 10.1080/00224065.2020.1848366 is OK
- 10.21105/joss.01686 is OK
- 10.1007/978-0-387-98141-3 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by moorepants** [comment]
@whedon check repository

**Comment by whedon** [comment]
Wordcount for `paper.md` is 1547

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.06 s (871.4 files/s, 138374.0 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
SVG                             19              0              0           3446
Markdown                         8            396              0           1391
R                               18            257            810           1013
Rmd                              3            327            571            289
YAML                             6             37              1            266
TeX                              1              9              1             73
Bourne Shell                     1              1              4              1
-------------------------------------------------------------------------------
SUM:                            56           1027           1387           6479
-------------------------------------------------------------------------------


Statistical information for the repository '46ca8f2d5064c97c0f925280' was
gathered on 2021/07/19.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Luke W. Johnston                 1             0            367           36.48
ismayc                           3           187              2           18.79
rudeboybert                      7           316            134           44.73

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
```

**Comment by rudeboybert** [comment]
Hey @moorepants:
- I just updated the `moderndive` package on [CRAN](https://cran.r-project.org/package=moderndive) to version 0.5.2
- I've tagged the release on GitHub [here](https://github.com/moderndive/moderndive/releases). By tag string I believe you mean [`v0.5.2`](https://github.com/moderndive/moderndive/tags)
- I created a [Zenodo archive](https://zenodo.org/record/5119261) of the release. The DOI is 10.5281/zenodo.5119261

**Comment by moorepants** [comment]
@whedon set v0.5.2 as version

**Comment by whedon** [comment]
OK. v0.5.2 is the version.

**Comment by moorepants** [comment]
@whedon set 10.5281/zenodo.5119261 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5119261" target="_blank">10.5281/zenodo.5119261</a> is the archive.

**Comment by moorepants** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00115/jose.00115/10.21105.jose.00115.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00115/jose.00115/10.21105.jose.00115.pdf) :page_facing_up: :point_left:

**Comment by moorepants** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.5070/T573020084 is OK
- 10.1080/00224065.2020.1848366 is OK
- 10.21105/joss.01686 is OK
- 10.1007/978-0-387-98141-3 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/jose-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/jose-papers/pull/61

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/jose-papers/pull/61, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by rudeboybert** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
I'm sorry @rudeboybert, I'm afraid I can't do that. That's something only editor-in-chiefs are allowed to do.

**Comment by moorepants** [comment]
@labarba This paper is ready to be accepted!

@lisamr and @lwjohnst86 thank you very much for your quality review. We at JOSE value your contribution.

@rudeboybert thank you for your patience with this submission. I apologize for losing track of it originally. The paper and software look great and I hope you are please with the final reviewed result. This is a very nice addition to the literature. Thanks for submitting to JOSE.

**Comment by rudeboybert** [comment]
Thank you @labarba and @moorepants for your work sustaining this wonderful initiative. I added my name to your list of potential reviewers awhile back, I hope to contribute someday.

@lisamr and @lwjohnst86 thanks once again for your detailed and thoughtful reviews. It was a pleasure working with you.

**Comment by moorepants** [comment]
> I added my name to your list of potential reviewers awhile back, I hope to contribute someday.

Good to know. We'll reach out for sure.

**Comment by lwjohnst86** [comment]
Congrats! :partying_face: and likewise :relaxed:

**Comment by labarba** [comment]
hi  @rudeboybert 👋  — Could you edit the metadata of the Zenodo archive so the title and author list matches the JOSE paper?

**Comment by rudeboybert** [comment]
Hi @labarba, done: https://zenodo.org/record/5119261

**Comment by labarba** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by labarba** [comment]
@openjournals/dev — is `whedon` having a bad day?

**Comment by whedon** [comment]
PDF failed to compile for issue #115 with the following error: 
```
 /app/vendor/bundle/ruby/2.6.0/gems/rest-client-2.1.0/lib/restclient/abstract_response.rb:249:in `exception_with_response': 500 Internal Server Error (RestClient::InternalServerError)
	from /app/vendor/bundle/ruby/2.6.0/gems/rest-client-2.1.0/lib/restclient/abstract_response.rb:129:in `return!'
	from /app/vendor/bundle/ruby/2.6.0/gems/rest-client-2.1.0/lib/restclient/request.rb:836:in `process_result'
	from /app/vendor/bundle/ruby/2.6.0/gems/rest-client-2.1.0/lib/restclient/request.rb:743:in `block in transmit'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/net/http.rb:920:in `start'
	from /app/vendor/bundle/ruby/2.6.0/gems/rest-client-2.1.0/lib/restclient/request.rb:727:in `transmit'
	from /app/vendor/bundle/ruby/2.6.0/gems/rest-client-2.1.0/lib/restclient/request.rb:163:in `execute'
	from /app/vendor/bundle/ruby/2.6.0/gems/rest-client-2.1.0/lib/restclient/request.rb:63:in `execute'
	from /app/vendor/bundle/ruby/2.6.0/gems/rest-client-2.1.0/lib/restclient.rb:66:in `get'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-b63fc70cc085/lib/whedon/compilers.rb:119:in `pdf_from_markdown'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-b63fc70cc085/lib/whedon/compilers.rb:13:in `generate_pdf'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-b63fc70cc085/lib/whedon/processor.rb:99:in `compile'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-b63fc70cc085/bin/whedon:88:in `compile'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-b63fc70cc085/bin/whedon:131:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `<main>'

```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
I'm sorry @arfon, I'm afraid I can't do that. That's something only editor-in-chiefs are allowed to do.

**Comment by arfon** [comment]
@labarba - could you try again now?

**Comment by labarba** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSE!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/jose-papers/pull/62
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/jose.00115](https://doi.org/10.21105/jose.00115)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by labarba** [comment]
Congratulations, @rudeboybert, your JOSE paper is published! 🚀 
(And thank you for your patience.)

Huge thanks to our Editor: @moorepants, and the Reviewers: @lisamr, @lwjohnst86 — your contributions make this possible 🙏

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://jose.theoj.org/papers/10.21105/jose.00115/status.svg)](https://doi.org/10.21105/jose.00115)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/jose.00115">
  <img src="https://jose.theoj.org/papers/10.21105/jose.00115/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://jose.theoj.org/papers/10.21105/jose.00115/status.svg
   :target: https://doi.org/10.21105/jose.00115
```
This is how it will look in your documentation:

[![DOI](https://jose.theoj.org/papers/10.21105/jose.00115/status.svg)](https://doi.org/10.21105/jose.00115)

**We need your help!**

Journal of Open Source Education is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. 
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-jose

**Comment by labarba** [comment]
:loudspeaker:  New Reviewer Sign-up Form!
https://forms.gle/Rizd3TcHnQKhrbYY7
